src/FluentMigrator.Extensions.Postgres/Postgres/PostgresExtensions.UsingIndexAlgorithm.cs

```diff
@@ -27,53 +26,99 @@ namespace FluentMigrator.Postgres
+            }
+
             var additionalFeatures = expression as ISupportAdditionalFeatures;
             additionalFeatures.Using(Algorithm.Hash);
-            return expression;
+            return new CreateHashIndexOptionSyntax(expression);
         }
 
-        public static ICreateIndexOptionsSyntax UsingGist(this ICreateIndexOptionsSyntax expression)
+        public static ICreateGiSTIndexOptionsSyntax UsingGist(this ICreateIndexOptionsSyntax expression)
         {
+            if (expression is ICreateIndexMethodOptionsSyntax)
+            {
+                throw new InvalidOperationException("Only can have one index method.");
+            }
+
             var additionalFeatures = expression as ISupportAdditionalFeatures;
             additionalFeatures.Using(Algorithm.Gist);
-            return expression;
+            return new CreateGistIndexOptionsSyntax(expression);
         }
 
-        public static ICreateIndexOptionsSyntax UsingSpgist(this ICreateIndexOptionsSyntax expression)
+        public static ICreateSpgistIndexOptionsSyntax UsingSpgist(this ICreateIndexOptionsSyntax expression)
         {
+            if (expression is ICreateIndexMethodOptionsSyntax)
+            {
+                throw new InvalidOperationException("Only can have one index method.");
+            }
+
             var additionalFeatures = expression as ISupportAdditionalFeatures;
             additionalFeatures.Using(Algorithm.Spgist);
-            return expression;
+            return new CreateSpgistIndexOptionsSyntax(expression);
         }
 
-        public static ICreateIndexOptionsSyntax UsingGin(this ICreateIndexOptionsSyntax expression)
+        public static ICreateGinIndexOptionsSyntax UsingGin(this ICreateIndexOptionsSyntax expression)
         {
+            if (expression is ICreateIndexMethodOptionsSyntax)
+            {
+                throw new InvalidOperationException("Only can have one index method.");
+            }
+
             var additionalFeatures = expression as ISupportAdditionalFeatures;
             additionalFeatures.Using(Algorithm.Gin);
-            return expression;
+            return new CreateGinIndexOptionsSyntax(expression);
         }
 
-        public static ICreateIndexOptionsSyntax UsingBrin(this ICreateIndexOptionsSyntax expression)
+        public static ICreateBrinIndexOptionsSyntax UsingBrin(this ICreateIndexOptionsSyntax expression)
         {
             var additionalFeatures = expression as ISupportAdditionalFeatures;
             additionalFeatures.Using(Algorithm.Brin);
-            return expression;
+            return new CreateBrinIndexOptionsSyntax(expression);
         }
 
-        public static ICreateIndexOptionsSyntax Using(this ICreateIndexOptionsSyntax expression, Algorithm algorithm)
+        public static ICreateIndexMethodOptionsSyntax Using(this ICreateIndexOptionsSyntax expression, Algorithm algorithm)
```

Comment:
This latest PR looks great.  You are really fast.

I'm going to play around with the syntax changes to make sure the tree works the right way and then merge it if I find no issues.